genienlp/models/transformer_sequence_classification.py

```diff
@@ -0,0 +1,104 @@
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import logging
+
+from transformers import AutoConfig, AutoModelForTokenClassification
+
+from ..data_utils.numericalizer import TransformerNumericalizer
+from ..models.base import GenieModel
+from ..util import adjust_language_code
+
+logger = logging.getLogger(__name__)
+
+
+class TransformerForSequenceClassification(GenieModel):
+    def __init__(self, config=None, *inputs, args, tasks, vocab_sets, save_directory=None, **kwargs):
+
+        num_labels = 0
+        if args.num_labels is not None:
+            num_labels = args.num_labels
+        else:
+            for task in tasks:
+                # if having multiple tasks choose max num_labels
+                if hasattr(task, 'num_labels'):
+                    num_labels = max(num_labels, task.num_labels)
+
+        config = AutoConfig.from_pretrained(
+            args.pretrained_model, cache_dir=args.embeddings, num_labels=num_labels
+        )
+        GenieModel.__init__(self, config)
+        self.args = args
+        if hasattr(config, 'd_model'):
+            args.dimension = config.d_model
+        else:
+            args.dimension = config.hidden_size
+
+        self.src_lang, self.tgt_lang = adjust_language_code(
+            config, args.pretrained_model, kwargs.get('src_lang', 'en'), kwargs.get('tgt_lang', 'en')
+        )
+
+        if save_directory is not None:
+            self.model = AutoModelForSequenceClassification.from_config(config)
+        else:
+            self.model = AutoModelForSequenceClassification.from_pretrained(
+                self.args.pretrained_model, cache_dir=self.args.embeddings, config=config
+            )
+
+        self.numericalizer = TransformerNumericalizer(
+            self.args.pretrained_model,
+            args,
+            max_generative_vocab=None,
+            save_dir=save_directory,
+            config=config,
+            src_lang=self.src_lang,
+            tgt_lang=self.tgt_lang,
+            vocab_sets=vocab_sets,
+            tasks=tasks,
+        )
+
+        self.model.resize_token_embeddings(self.numericalizer.num_tokens)
+
+        self.numericalizer.answer_pad_id = -100
+
+    def add_new_vocab_from_data(self, tasks, resize_decoder=False):
+        super().add_new_vocab_from_data(tasks, resize_decoder)
+        self.model.resize_token_embeddings(self.numericalizer.num_tokens)
+
+    def forward(self, *input, **kwargs):
+        if self.training:
+            batch = input[0]
+            outputs = self.model(
+                batch.context.value,
+                labels=batch.answer.value,
+                attention_mask=(batch.context.value != self.numericalizer.pad_id),
+            )
+            return outputs
+        else:
+            return self.model(**kwargs)
```

Comment:
I'm not sure this if-else is correct.
The training part looks right, but for the inference part I think you need to write a new branch in [generate_with_model](https://github.com/stanford-oval/genienlp/blob/master/genienlp/validate.py#L42) and make sure whatever arguments you pass there are passed here.

Maybe @Mehrad0711 can confirm here.

Reply:
The model code looks correct but [this if statement](https://github.com/stanford-oval/genienlp/blob/6e45a5699d4491975a2385930383d26ecd9eb321/genienlp/validate.py#L55) should include `TransformerForSeqeunceClassification` as well.  I think the `generate_with_classification_model` should work for sequence classification task as well, but you need to run the code and make sure that's the case.